src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -1979,6 +1979,44 @@ public void actionPerformed(final ActionEvent e) {
     }
   };
 
+  /**
+   * Prompt user for which producer territory (factory) to use when producing units in sea
+   * zones that have multiple adjacent factories.
+   * 
+   * @param candidates
+   *        - list of producer territories to choose from
+   * @param unitTerritory
+   *        - territory units are being placed
+   * @return territory selected to use for production
+   */
+  public Territory selectProducerTerritoryForUnits(final Collection<Territory> candidates,
+      final Territory unitTerritory) {
+    if (messageAndDialogThreadPool == null) {
+      return null;
+    }
+    messageAndDialogThreadPool.waitForAll();
+    mapPanel.centerOn(unitTerritory);
+    final AtomicReference<Territory> selected = new AtomicReference<>();
+    SwingAction.invokeAndWait(() -> {
+      final JList<Territory> list = new JList<>(new Vector<>(candidates));
+      list.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+      list.setSelectedIndex(0);
+      final JPanel panel = new JPanel();
+      panel.setLayout(new BorderLayout());
+      final JScrollPane scroll = new JScrollPane(list);
+      panel.add(scroll, BorderLayout.CENTER);
+      panel.add(BorderLayout.NORTH, new JLabel("Producer territory for units in " + unitTerritory.getName()));
+      final String[] options = {"OK", "Canel"};
```

Comment:
`"Canel"` --> `"Cancel"`

Note, however, that passing `null` for the `options` parameter will use the appropriate locale-specific button labels.  The Javadocs for `JOptionPane#showOptionDialog()` seem to imply this is only done if `optionType` is `YES_NO_OPTION` or `YES_NO_CANCEL_OPTION`, but it appears we've used `null` in other locations where `OK_CANCEL_OPTION` is specified (e.g. `SetPasswordAction#actionPerformed()`).